controllers/storageclass/storageclass_controller_test.go

```diff
@@ -216,6 +216,118 @@ var _ = Describe("StorageClassWatcher", func() {
+			ctx := context.TODO()
+
+			By("By querying the created StorageClass")
+			scLookupKey := types.NamespacedName{
+				Name:      storageClassName,
+				Namespace: "",
+			}
+			createdSc := &storagev1.StorageClass{}
+
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, scLookupKey, createdSc)
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
+
+			By("By querying the ConfigMap")
+			cmLookupKey := types.NamespacedName{
+				Name:      util.PoolConfigmapName,
+				Namespace: namespace,
+			}
+			createdCm := &corev1.ConfigMap{}
+
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, cmLookupKey, createdCm)
+				if err == nil {
+					var sp util.FlashSystemClusterMapContent
+					err = json.Unmarshal([]byte(createdCm.Data[FlashSystemName]), &sp)
+					if err == nil {
+						return sp.ScPoolMap[storageClassName] == poolName
+					}
+				}
+				return false
+			}, timeout, interval).Should(BeTrue())
+
+			By("By querying the FlashSystemCluster")
+			fscLookupKey := types.NamespacedName{
+				Name:      FlashSystemName,
+				Namespace: namespace,
+			}
+			createdFsc := &odfv1alpha1.FlashSystemCluster{}
+
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, fscLookupKey, createdFsc)
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
+
+			Expect(createdFsc.Name).To(Equal(FlashSystemName))
+			Expect(createdFsc.Namespace).To(Equal(namespace))
+			Expect(createdFsc.Spec.Secret.Name).To(Equal(secretName))
+
+			By("creating a new secret with topology awareness")
+			topologySecretName := "topology-secret"
+			secret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      topologySecretName,
+					Namespace: namespace,
+				},
+				Data: map[string][]byte{
+					"config": []byte("IHsKICAgImRlbW8tbWFuYWdlbWVudC1pZC0xIjogewogICAgICJ1c2VybmFtZSI6ICJkZW1vLXVzZX" +
+						"JuYW1lLTEiLAogICAgICJwYXNzd29yZCI6ICJkZW1vLXBhc3N3b3JkLTEiLAogICAgICJtYW5hZ2VtZW50X2FkZHJlc" +
+						"3MiOiAiZGVtby1tYW5hZ2VtZW50LWFkZHJlc3MtMSIsCiAgICAgInN1cHBvcnRlZF90b3BvbG9naWVzIjogWwogICAg" +
+						"ICAgewogICAgICAgICAidG9wb2xvZ3kuYmxvY2suY3NpLmlibS5jb20vZGVtby1yZWdpb24iOiAiZGVtby1yZWdpb24" +
+						"tMSIsCiAgICAgICAgICJ0b3BvbG9neS5ibG9jay5jc2kuaWJtLmNvbS9kZW1vLXpvbmUiOiAiZGVtby16b25lLTEiCi" +
+						"AgICAgICB9CiAgICAgXQogICB9LAogICAiZGVtby1tYW5hZ2VtZW50LWlkLTIiOiB7CiAgICAgInVzZXJuYW1lIjogI" +
+						"mRlbW8tdXNlcm5hbWUtMiIsCiAgICAgInBhc3N3b3JkIjogImRlbW8tcGFzc3dvcmQtMiIsCiAgICAgIm1hbmFnZW1l" +
+						"bnRfYWRkcmVzcyI6ICJkZW1vLW1hbmFnZW1lbnQtYWRkcmVzcy0yIiwKICAgICAic3VwcG9ydGVkX3RvcG9sb2dpZXM" +
+						"iOiBbCiAgICAgICB7CiAgICAgICAgICJ0b3BvbG9neS5ibG9jay5jc2kuaWJtLmNvbS9kZW1vLXJlZ2lvbiI6ICJkZW" +
+						"1vLXJlZ2lvbi0yIiwKICAgICAgICAgInRvcG9sb2d5LmJsb2NrLmNzaS5pYm0uY29tL2RlbW8tem9uZSI6ICJkZW1vL" +
+						"XpvbmUtMiIKICAgICAgIH0KICAgICBdCiAgIH0KIH0="),
+				},
+			}
+			Expect(k8sClient.Create(ctx, secret)).Should(Succeed())
+
+			By("By creating a new topology StorageClass")
```

Comment:
Not sure what we are trying to test here.
The idea is that we need to assume the objects were created successfully, and now we want to verify all other parts related to us are performed correctly (like adding the StorageClass to all relevant StorageSystem entries with the correct pool name, etc.)